app/models/spree/stock/estimator_decorator.rb

```diff
@@ -3,7 +3,10 @@ module Stock
     module EstimatorDecorator
       def shipping_rates(package, shipping_method_filter = ShippingMethod::DISPLAY_ON_FRONT_END)
 
-        if package.use_easypost?
+        if package.use_easypost? && 
+          ((shipping_method_filter == ShippingMethod::DISPLAY_ON_FRONT_END) && 
+                    Spree::Config.use_easypost_on_frontend)
+
           shipment = package.easypost_shipment
```

Comment:
make this nested conditional a private method with a meaningful name like 

```
if should_calculate_with_easypost?
  # things here 
end 

private 

def should_calculate_with_easypost?
  package.use_easypost? &&
  ((shipping_method_filter == ShippingMethod::DISPLAY_ON_FRONT_END) && Spree::Config.use_easypost_on_frontend)
end

Reply:
going to hold on this for now